File: insertion-sort/python/insertion_sort.py
def insertion_sort(the_list):
    if len(the_list) < 2:
        return the_list
    for i in range(len(the_list)):
        j = i
        while j > 0 and the_list[j] < the_list[j-1]:
            j_value = the_list[j]
            the_list[j] = the_list[j-1]
            the_list[j- 1] = j_value
            j -= 1
    return the_list

a_list = [4, 6, 2, 3, 8, 5]
print(a_list)
insertion_sort(a_list)
print(a_list)

File: insertion-sort/javascript/insertionSort.js
Array.prototype.insertionSort = function(){
  if(this.length < 2){
    return this;
  }
  for(var i=0; i<this.length; i++){
    var j = i;
    while(j > 0 && this[j] < this[j-1]){
      var jValue = this[j];
      this[j] = this[j-1];
      this[j-1] = jValue;
      j -= 1;
    }
  }
  return this;
}

var aList = [8, 3, 5, 1, 3, 9, 7];
aList.insertionSort();
console.log(aList);
